plugins/org.python.pydev.ast/src/org/python/pydev/ast/interpreter_managers/InterpreterInfo.java

```diff
@@ -1897,7 +1899,24 @@ public void setStringSubstitutionVariables(Properties stringSubstitutionOriginal
 
     @Override
     public Properties getStringSubstitutionVariables() {
-        return this.stringSubstitutionVariables;
+        return getStringSubstitutionVariables(false);
+    }
+
+    @Override
+    public Properties getStringSubstitutionVariables(boolean addEnvironmentVaiables) {
+        Properties ret = new Properties();
+        ret.putAll(this.stringSubstitutionVariables);
+        if (addEnvironmentVaiables) {
+            Map<String, String> formattedEnv = new Hashtable<String, String>();
```

Comment:
Prefer a `HashMap` instead of a `HashTable` here (the difference is that `HashTable` is synchronized and thus slower -- use it only if you actually plan to access it from several threads).

Although in this particular case, why don't you just put it in `ret` directly?

Reply:
Yup, put it in `ret` directly makes a lot more sense